src/vcHTTP.cpp

```diff
@@ -0,0 +1,318 @@
+    char *pGetStart = nullptr;
+
+    while (*pCarat != ' ' && *pCarat != '\0' && *pCarat != '\r' && *pCarat != '\n')
+    {
+      if (pGetStart == nullptr && *pCarat == '?')
+      {
+        pGetStart = pCarat;
+        *pGetStart = '\0';
+      }
+
+      ++pCarat;
+    }
+
+    if (*pCarat == ' ')
+    {
+      *pCarat = '\0';
+
+      // Clip extra slashes
+      while (*pURIStart && *pURIStart == '/')
+        pURIStart++;
+      udStrcat(pHeaders->URI, pURIStart);
+
+      if (pGetStart != nullptr)
+      {
+        pHeaders->tokens.Set("Get = {}");
+
+        udJSON *pGetToken;
+        pHeaders->tokens.Get(&pGetToken, "Get");
+
+        vcHTTP_SplitAndInject(pGetToken, &pGetStart[1], '&', '=', false);
+
+        *pGetStart = '?'; //Reset the character
+      }
+
+      *pCarat = ' ';
+    }
+    else
+    {
+      pHeaders->status = vcHTTP_BadRequest;
+    }
+  }
+  else
+  {
+    pHeaders->status = vcHTTP_BadRequest;
+  }
+
+  //Extract the rest of the headers
+  do
+  {
+    char *pLineStart = pCarat;
+    char *pColon = nullptr;
+    char *pLineEnd = nullptr;
+
+    while (*pCarat != '\r' && *pCarat != '\n' && *pCarat != '\0')
+    {
+      if (*pCarat == ':' && pColon == nullptr)
+        pColon = pCarat;
+      ++pCarat;
+    }
+
+    pLineEnd = pCarat;
+
+    if (pLineEnd - pLineStart < 2)
+      break; //Finished processing
+
+    if (pColon != nullptr)
+    {
+      //Null terminate strings
+      char endToken = *pLineEnd;
+      *pColon = '\0';
+      *pLineEnd = '\0';
+
+      char *pColonStart = pColon + 1;
+
+      while ((*pColonStart == ' ' || *pColonStart == '\t') && pColonStart < pLineEnd)
+        ++pColonStart;
+
+      pHeaders->tokens.Set("%s = '%s'", pLineStart, pColonStart);
+
+      if (udStrEqual(pLineStart, "User-Agent"))
+        pHeaders->pUserAgentString = pHeaders->tokens.Get("User-Agent").AsString(); //The string is owned by the value
+
+      //Remove null terminators
+      *pLineEnd = endToken;
+      *pColon = ':';
+    }
+
+    //Remove extra whitespace
+    int total = 0;
+    while (*pCarat == '\r' || *pCarat == '\n')
+    {
+      ++pCarat;
+      ++total;
+    }
+
+    //Exit if the body's been reached, signified by "\r\n\r\n", or if the end of the header has been reached and there's no body, signified by '\0'.
+    if (*pCarat == '\0' || total == 4)
+      break;
+  } while (true);
+
+  //Copy body pointer out
+  if (ppBodyData != nullptr && *pCarat != '\0')
+    *ppBodyData = pCarat;
+
+  //Handle the Cookies strings
+  if (pHeaders->tokens.Get("Cookie").IsString())
+  {
+    udJSON *pCookieValue;
+    pHeaders->tokens.Get(&pCookieValue, "Cookie");
+
+    char *pCookieStr;
+    pCookieValue->ExtractAndVoid((const char**)&pCookieStr);
+
+    vcHTTP_SplitAndInject(&pHeaders->cookies, pCookieStr, ';', '=', true);
+
+    udFree(pCookieStr);
+  }
+
+  if (!pHeaders->tokens.Get("Cookie").IsObject())
+    pHeaders->tokens.Set("Cookie = {}");
+}
+
+const char* vcHTTP_StatusAsText(vcHTTPStatus status)
```

Comment:
@OsiJr This function isn't used and I've made two comments against it, should vcHTTP contain so much unused functionality?